packages/menu/src/Menu.tsx

```diff
@@ -88,46 +89,152 @@ function Menu({
   ...rest
 }: MenuProps) {
   const refs: Array<HTMLElement> = [];
+  const titleArr: Array<string> = [];
+
   const hasSetInitialFocus = useRef(false);
+  const hasSetInitialOpen = useRef(false);
+  // @ts-ignore: using this setState to trigger a subMenu close after animation
+  const [closed, setClosed] = useState(false); // eslint-disable-line
+  const [currentSubMenu, setCurrentSubMenu] = useState<React.ReactElement<
+    HTMLLIElement
+  > | null>(null);
   const { setUsingKeyboard } = useUsingKeyboardContext();
 
-  const updatedChildren = React.Children.map(children, child => {
-    if (
-      (isComponentType(child, 'MenuItem') && !child.props.disabled) ||
-      isComponentType(child, 'FocusableMenuItem')
-    ) {
-      return React.cloneElement(child, {
-        ref: (ref: HTMLElement) => {
-          refs.push(ref);
-          if (
-            open &&
-            refs.length === 1 &&
-            hasSetInitialFocus.current === false
-          ) {
-            refs[0].focus();
-            setUsingKeyboard(true);
-            setFocused(refs[0]);
-            hasSetInitialFocus.current = true;
-          }
-        },
-        onFocus: (e: Event) => setFocused(e.target as HTMLElement),
-      });
-    }
+  function updateChildren(
+    children: Array<React.ReactElement>,
+  ): Array<React.ReactElement> {
+    return React.Children.map(children, (child: React.ReactElement) => {
+      if (child.props?.disabled) {
+        return child;
+      }
 
-    return child;
-  });
+      let currentChildRef: HTMLElement | null = null;
+
+      const setRef = (ref: HTMLElement) => {
+        if (ref == null) {
+          return;
+        }
+
+        refs.push(ref);
+        currentChildRef = ref;
+
+        if (open && hasSetInitialFocus.current === false) {
+          setFocus(refs[0]);
+          // This allows the focus state to be shown via the LeafyGreen Provider
+          setUsingKeyboard(true);
+          hasSetInitialFocus.current = true;
+        }
+      };
+
+      const onFocus = ({ target }: { target: HTMLElement }) => {
+        setFocused(target);
+      };
+
+      if (isComponentType<'SubMenu'>(child, 'SubMenu')) {
+        if (titleArr.includes(child.props.title)) {
```

Comment:
Boy we use `child.props.title` quite a bit. Might be worth destructuring props to make the code more succinct.

Reply:
👍